recipes-extended/xen/files/libxl-add-cores-per-socket-support.patch

```diff
@@ -0,0 +1,77 @@
+################################################################################
+SHORT DESCRIPTION:
+################################################################################
+Cores-per-socket support in libxl
+
+################################################################################
+LONG DESCRIPTION:
+################################################################################
+The OpenXT Xen patchqueue adds a DOMCTL call for new function that consolidates
+the number of cores on the socket that is exposed to the guest.  This is useful
+for OS's that have a limit on the number of visible sockets. In OpenXT, this
+the the method by which we assign more than 2 vcpus to Windows guests (win 7
+in particular).
+
+################################################################################
+CHANGELOG
+################################################################################
+Authors:
+Chris Rogers <rogersc@ainfosec.com>
+
+################################################################################
+REMOVAL
+################################################################################
+Do not remove
+
+################################################################################
+UPSTREAM PLAN
+################################################################################
+This is specific to OpenXT, nothing to upstream here.
+
+################################################################################
+INTERNAL DEPENDENCIES
+################################################################################
+
+################################################################################
+PATCHES
+################################################################################
+
+Index: xen-4.6.4/tools/libxl/libxl_dom.c
+===================================================================
+--- xen-4.6.4.orig/tools/libxl/libxl_dom.c
++++ xen-4.6.4/tools/libxl/libxl_dom.c
+@@ -370,6 +370,8 @@ int libxl__build_pre(libxl__gc *gc, uint
```

Comment:
The context of this hunk is invalid against libxl's code (without --fuzz=0 it will actually apply that hunk with ~1700 line of fuzz).
This hunk has 2 contiguous empty lines in its context (patch will look for that in the original file, there are 2 contiguous empty lines in libxl_userdata_retrieve) between which the 2 added lines will be added. I think the intent was to add the if statement on 2 lines plus one empty line following it.